### PR TITLE
Filter out UDP candidates from SDP if allowing only TCP.

### PR DESCRIPTION
**Pull request by boks1971**

### CODE REVIEW

File: pkg/rtc/transport_test.go [hunk 1]
```diff
@@ -269,6 +271,70 @@ func TestNegotiationFailed(t *testing.T) {
...
+		}
+		return numUDPCandidates
+	}
+	require.NotZero(t, getNumUDPCandidates(parsed))
+
+	transport.SetPreferTCP(true)
+	filteredOffer = transport.FilterCandidates(offer)
+	parsed, err = filteredOffer.Unmarshal()
+	require.NoError(t, err)
+	require.Zero(t, getNumUDPCandidates(parsed))
```
Comment (boks1971) [inline pkg/rtc/transport_test.go:335]: Wanted to actually add TCP candidates and ensure that it is not filtered. But, seems like setting that up in a test is a bit laborious. Maybe, I can just add a couple of hard coded TCP candidates?

Comment (davidzhao) [reply pkg/rtc/transport_test.go:335]: sg for the test.

**Review by davidzhao — APPROVED** [review]
lgtm!

File: pkg/rtc/transport_test.go (hunk 1 above)
Comment (boks1971) [reply pkg/rtc/transport_test.go]: Actually, went ahead and added canned candidates and testing with it. Just to be safe.

**Pull request merged**